### PR TITLE
Improved Supplier Service Decorator address handling

### PR DESCRIPTION
**Pull request by nvindice**
See #129

### CODE REVIEW

**Comment by aimeos** [comment]
Thank you very much for your PR! :-)

I think the code could be more simplified by using only one additional service attribute named `supplier.address`. What is displayed and added to the service as attribute can be defined by the local translation of the address string where everything can be left out that is undesired. Also, only `supplier.display-code` needs to be a configuration option because then, everything else is unnecessary.

What do you think?

File: lib/mshoplib/src/MShop/Service/Provider/Decorator/Supplier.php [hunk 1]
```diff
@@ -139,4 +190,39 @@ public function getConfigFE( \Aimeos\MShop\Order\Item\Base\Iface $basket )
...
+			}
+			
+			if( $this->getConfigValue('supplier.display-code') === 0 ) {
+				// remove code attribute for summary page / customer email
+				unset( $attributes['supplier.code'] );
+			}
+			
+		}
+
+		$this->setAttributes( $orderServiceItem, $attributes, 'delivery' );
```
Comment (aimeos) [inline lib/mshoplib/src/MShop/Service/Provider/Decorator/Supplier.php]: This is wrong, you prevent all further decorators to modify the service attributes as well. Instead, you need to add your attributes (or remove existing ones) and call the next decorator using `$this->getProvider()->setConfigFE(...)`

Comment (nvindice) [reply lib/mshoplib/src/MShop/Service/Provider/Decorator/Supplier.php]: Sorry, didn't know that. I fixed it!

**Comment by aimeos** [comment]
Don't know if someone would like to show the code only. Up to now this was only displayed because your code was missing ;-)

**Comment by nvindice** [comment]
Great! What is the desired default behaviour? Address only/code optional, or both displayed by default?

**Comment by aimeos** [comment]
I think so too :-)

**Comment by aimeos** [comment]
There's one error for PHP 5.4 you should fix too. Please have a look into the Travis build

**Comment by nvindice** [comment]
Should be fixed and ready for merging.

**Comment by aimeos** [comment]
The short address is now always added and it's impossible to add an address detail like the state for US customers. I would suggest to use the full address only because if address details are not set, multiple line breaks and white spaces are removed so only the relevant details will be shown in a nice format, e.g. for label, company, address1, address2, postal and city would be shown as
```
label
company
address1 address2
postal city
```
This is also what you've tried to achive with the short version.

**Comment by nvindice** [comment]
In our use case it's not possible to use `address` at both places, because the used formats differ heavily. IMO it should be possible to define different formats for delivery and summary page/e-mail (customizable via translation). I'd suggest to leave it there as `short` but add all possibly needed information. Then every user can decide which information he wants to display. IMO a comma separated string makes a lot of sense here, so the summary page won't bloat.

Another possibility would be to use another translation domain eg. `mshop/summary`, but this might be a rather dirty solution...

What do you think?

**Comment by aimeos** [comment]
I wouldn't use another translation domain too but we could use another string like
```
$context->getI18n()->dt( 'mshop', '%1$s, %2$s, %3$s %4$s, %6$s %7$s')
```
Then, we could feed in the same data (%1$s to %13%s) and get different translatable results. I would also remove the check for address1 and city because other users may want to have a short version even if these data isn't available in their installation

**Comment by nvindice** [comment]
Totally agreeing on this, I will commit the final solution in a minute!

**Comment by aimeos** [comment]
Thanks a lot! :-)

**Pull request merged**